src/remote-backend/httpfunc.cc

```diff
@@ -24,4 +24,24 @@ void RemoteBackend::notFound(const Rest::Request &request, Http::ResponseWriter
   sendError(response, string(), Http::Code::Not_Found);
 }
 
+std::string RemoteBackend::urlDecode(std::string& eString) {
+  // from http://www.cplusplus.com/forum/general/94849/
+  std::string ret;
+  char ch;
+  size_t i;
+  unsigned int j;
+  for (i = 0; i < eString.length(); i++) {
+    if (int(eString[i]) == 37) {
+      sscanf(eString.substr(i + 1, 2).c_str(), "%x", &j);
```

Comment:
If `eString` ends in `%8`, this will put `8` in `j`

Reply:
`%xy` (i.e. no valid hex) will leave `j` silently unchanged

Reply:
As the pdns_server is trusted, this is fine for now